### PR TITLE
Feature eslint fix violations

### PR DESCRIPTION
**Pull request by angel4576**
# Description

- Fixed all ESLint violations for most of the js files
- Configured ESLint rules in .eslintrc.json
- Added .eslintignore to make ESLint ignore some files.  
#567 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
`npm run build`

**My Configuration**:
* `npm --version`: 8.3.1
* `node --version`: v16.14.0
* `python3 --version`: 3.10.0
* OS: Windows
* ARCH: x86_64

### CODE REVIEW

**Comment by dkoes** [comment]
Checks need to pass before this can be considered for merging.

**Review by tjakubiec218 — APPROVED** [review]

File: 3Dmol/3dmol.js [hunk 1]
```diff
@@ -304,369 +298,379 @@ $3Dmol.base64ToArray = function(base64) {
...
+      !Number.isFinite(atom.y) ||
+      !Number.isFinite(atom.z)
+    )
+      continue;
+    cnt += 1;
+    xsum += atom.x;
+    ysum += atom.y;
+    zsum += atom.z;
+
+    xmin = xmin < atom.x ? xmin : atom.x;
```
Comment (dkoes) [inline 3Dmol/3dmol.js:593]: what eslint rule is requiring clarifying parentheses be removed?  as far as I can tell, it doesn't mind so why were these changes made?

File: 3Dmol/3dmol.js [hunk 2]
```diff
@@ -675,90 +679,87 @@ if( typeof(define) === 'function' && define.amd) {
...
-    var methods = Object.getOwnPropertyNames(this.glviewer1) //get all methods of glviewer object
-    .filter(function(property) {
-        return typeof that.glviewer1[property] == 'function';
-    });
+$3Dmol.createStereoViewer = function (element) {
+  const that = this;
+  element = typeof element == 'string' ? $(`#${element}`) : element;
+  if (!element) return;
+
+  const viewers = $3Dmol.createViewerGrid(element, {rows: 1, cols: 2, controlAll: true});
```
Comment (dkoes) [inline 3Dmol/3dmol.js:687]: no API changes

**Review by dkoes — CHANGES_REQUESTED** [review]
I went through one not very large file (3dmol.js) and found at least two breaking changes.  This does not inspire confidence.  I think perhaps large scale changes like this should wait until there is greater code coverage with the tests.  It does not help that there appear to be some changes made that aren't required by eslint, which just results in more diffs to review.

In any case, eslint should be run as part of the build process.  There are currently
✖ 19676 problems (11196 errors, 8480 warnings)
This PR cannot be merged until those are zero.

I'd recommend reducing the scope of the linting to just the eslint recommended rules, or an even smaller subset (e.g. disable the suggestions).  Once the system is integrated into the build it will be easy enough to incrementally enable rules.

File: 3Dmol/3dmol.js [hunk 3]
```diff
@@ -1,22 +1,20 @@
...
-    
-    return my;
+// eslint-disable-next-line no-var
+var $3Dmol = (function threeDeeMollContructor(window) {
+  const my = window.$3Dmol || {};
 
+  return my;
 })(window);
 
-if ( typeof module === "object" && typeof module.exports === "object" ) {
```
[the remark this answers is not in the document]
Comment (angel4576) [reply 3Dmol/3dmol.js:17]: Hello professor, the option for eqeqeq rule is set to "smart", which enforces the use of === and !== except for comparing two literal values, evaluating typeof values, and comparing with null

File: 3Dmol/3dmol.js [hunk 4]
```diff
@@ -25,96 +23,94 @@ if ( typeof module === "object" && typeof module.exports === "object" ) {
...
+  }
+  return null;
+});
+
 /**
  * Create and initialize an appropriate viewer at supplied HTML element using specification in config
  @function $3Dmol.createViewer
  * @param {Object | string} element - Either HTML element or string identifier
  * @param {ViewerSpec} config Viewer configuration
- * @param {Object} shared_viewer_resources shared resources between viewers' renderers
```
[the remark this answers is not in the document]
Comment (angel4576) [reply 3Dmol/3dmol.js:117]: I can fix that. Sorry about that

**Comment by gncs** [comment]
Using a linter like ESLint is an excellent idea in my opinion and I highly appreciate @angel4576 effort to introduce a consistent style to the project. At the same time, I understand @dkoes 's concern that other API and potentially behavioral changes were introduced in this PR.

As projects typically benefit from a consistent code style and adherence to best practices, I encourage the use of a linter. One could consider restarting this PR - maybe with a more minimalist linter config. As @dkoes said, one can always make the linter stricter or more pedantic later on.

What are your thoughts?

**Comment by dkoes** [comment]
@RyanGuild  is in the middle of a massive refactor that will convert all of 3Dmol.js to typescript with an updated build system that includes linting.  Hence all the merge conflicts.

**Comment by gncs** [comment]
Exciting, thank you for your efforts.

**Comment by GermaVinsmoke** [comment]
@dkoes By when we can expect to see the new Typescript change? Interested to contribute.

**Comment by GermaVinsmoke** [comment]
I see. Maybe I can go through the typescript branches created by @RyanGuild and see what I can do 🤔

**Comment by dkoes** [comment]
That would be fantastic!

**Comment by GermaVinsmoke** [comment]
It looks like, he was trying to add **rollup** first but then he stopped. Can be seen in [feature-modules](https://github.com/3dmol/3Dmol.js/tree/feature-modules) branch. Later on, he decided to use Parcel as a bundler.

Some of the branches which are not merged in the master are (Correct me if I'm wrong) -
- [tsport-partial-charges](https://github.com/3dmol/3Dmol.js/tree/tsport-partial-charges)
- [tsport-marchingcube](https://github.com/3dmol/3Dmol.js/tree/tsport-marchingcube)
- [tsport-webgl](https://github.com/3dmol/3Dmol.js/tree/tsport-webgl)
- [tsport-parsers](https://github.com/3dmol/3Dmol.js/tree/tsport-parsers)

I think we need to first see what all changes were made in these branches and whether these are even needed or not. Too much work! 😬

**Pull request closed without merging**